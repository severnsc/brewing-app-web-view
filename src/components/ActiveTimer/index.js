import React from "react"
import styles from "./styles"
import PropTypes from "prop-types"

const ActiveTimer = ({ name, time, isRunning, startTimer, stopTimer, resetTimer }) => {

	const startButton = <button onClick={startTimer} style={styles.start}>Start</button>
	const stopButton = <button onClick={stopTimer} style={styles.stop}>Stop</button>

	return (
		<div>
			<h2>{name}</h2>
			<h2>{time}</h2>
			{isRunning ? stopButton : startButton}
			<button onClick={resetTimer} style={styles.button}>Reset</button>
		</div>
	)
}

ActiveTimer.propTypes = {
	name: PropTypes.string.isRequired,
	time: PropTypes.string.isRequired,
	isRunning: PropTypes.bool.isRequired,
	startTimer: PropTypes.func.isRequired,
	stopTimer: PropTypes.func.isRequired,
	resetTimer: PropTypes.func.isRequired
}

export default ActiveTimer