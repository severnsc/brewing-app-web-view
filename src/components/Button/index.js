import React from "react"
import PropTypes from "prop-types"
import globalStyles from "../styles"

const Button = ({ onClick, value, style }) => (
	<button style={{...globalStyles.button, ...style}} onClick={onClick}>
		{value}
	</button>
)

Button.propTypes = {
	onClick: PropTypes.func.isRequired,
	value: PropTypes.oneOfType([
		PropTypes.string,
		PropTypes.number
	]).isRequired,
	style: PropTypes.object
}

export default Button