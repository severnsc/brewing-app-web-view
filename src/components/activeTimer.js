import React from "react"
import PropTypes from "prop-types"

const ActiveTimer = ({ time, isRunning, startTimer, stopTimer, resetTimer }) => {

	const styles = {
		button: {
			color: "white",
			backgroundColor: "grey"
		}
	}

	styles.start = {
		...styles.button,
		backgroundColor: "green"
	}

	styles.stop = {
		...styles.button,
		backgroundColor: "red"
	}

	const startButton = <button onClick={startTimer} style={styles.start}>Start</button>
	const stopButton = <button onClick={stopTimer} style={styles.stop}>Stop</button>

	return (
		<div>
			<h2>{time}</h2>
			{isRunning ? stopButton : startButton}
			<button onClick={resetTimer} style={styles.button}>Reset</button>
		</div>
	)
}

ActiveTimer.propTypes = {
	time: PropTypes.string.isRequired,
	isRunning: PropTypes.bool.isRequired,
	startTimer: PropTypes.func.isRequired,
	stopTimer: PropTypes.func.isRequired,
	resetTimer: PropTypes.func.isRequired
}

export default ActiveTimer