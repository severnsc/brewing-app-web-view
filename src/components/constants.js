export const errorRed = "hsl(353, 85%, 44%)"
export const green = "hsl(167, 90%, 38%)"