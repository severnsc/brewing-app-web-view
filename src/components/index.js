import Timer from "./Timer"
import ForgotPasswordForm from "./ForgotPasswordForm"
import HoverableTableRow from "./HoverableTableRow"
import InventoryItemForm from "./InventoryItemForm"
import Modal from "./Modal"
import NavBar from "./NavBar"
import ResetPasswordForm from "./ResetPasswordForm"
import SearchBar from "./SearchBar"
import SortableTable from "./SortableTable"
import TimerAlertForm from "./TimerAlertForm"
import TimerForm from "./TimerForm"
import InventoryForm from "./InventoryForm"
import FixedNewItemButton from "./FixedNewItemButton"
import NewTimerForm from "./NewTimerForm"
import NewTimerAlertForm from "./NewTimerAlertForm"
import {
	SignupForm,
	LoginForm
} from "./Forms"

export {
	Timer,
	ForgotPasswordForm,
	HoverableTableRow,
	InventoryItemForm,
	LoginForm,
	Modal,
	NavBar,
	ResetPasswordForm,
	SearchBar,
	SignupForm,
	SortableTable,
	TimerAlertForm,
	TimerForm,
	InventoryForm,
	FixedNewItemButton,
	NewTimerForm,
	NewTimerAlertForm
}