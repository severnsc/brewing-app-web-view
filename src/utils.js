import moment from "moment"
import convert from "convert-units"

export const constructErrorMessage = (name, type, item) => 
  `${name} must be of type ${type}! Got value: ` + item

export const convertMsToMinutesSecondsString = ms => {
	const tempTime = moment.duration(ms)
	let hours = tempTime.hours()
	let minutes = tempTime.minutes()
	let seconds = tempTime.seconds()
	let times = [hours, minutes, seconds]
	times = times.map(time => time < 10 ? "0" + time : time)
	return `${times[0]}:${times[1]}:${times[2]}`
}

export const convertMinutesSecondsStringToMs = string => {
	const splitTime = string.split(":")
	return splitTime[0] * 60 * 60 * 1000 + splitTime[1] * 60 * 1000 + splitTime[2] * 1000
}

export const weightUnits = type => {
	switch(type){
		case "metric":
			return "(kg, g)"

		default:
			return "(lbs, oz)"
	}
}

export const convertKgToLbs = kg => convert(kg).from("kg").to("lb")

export const convertLbsToKg = lbs => convert(lbs).from("lb").to("kg")

export const convertWeight = (weight, originalUnits, convertedUnits) => {
	if(originalUnits === "imperial" && convertedUnits === "metric"){
		return convertLbsToKg(weight)
	}

	if(originalUnits === "metric" && convertedUnits === "imperial"){
		return convertKgToLbs(weight)
	}

	return weight
}

const isInt = num => num % 1 === 0

export const formatLbsOzString = weight => {
	const lbs = Math.trunc(weight)
	const oz = convert(weight - lbs).from("lb").to("oz")
	const lbsString = lbs <= 1 ? "lb" : "lbs"
	return `${lbs} ${lbsString}, ${oz} oz`
}

export const formatKgGString = weight => {
	const kgs = Math.trunc(weight)
	let g = parseFloat(convert((weight - kgs)).from("kg").to("g").toFixed(1))
	if(isInt(g)) g = (g).toFixed(0)
	const kgsString = kgs <= 1 ? "kg" : "kgs"
	return `${kgs} ${kgsString}, ${g} g`
}

export const formatWeightString = (weight, units) => {
	if(units === "imperial"){
		return formatLbsOzString(weight)
	}

	if(units === "metric"){
		return formatKgGString(weight)
	}
}

export const SRMtoL = srm => (srm + 0.6) / 1.35
export const LtoSRM = l => (1.35 * l) - 0.6

export const convertColor = (from, to, value) => {
	if(from === "SRM" && to === "L"){
		return SRMtoL(value)
	}

	if(from === "L" && to === "SRM"){
		return LtoSRM(value)
	}

	return value
}

export const srmToRGB = srm => {
    // Returns an RGB value based on SRM
    var r=0, g=0, b=0;

    if (srm>=0 && srm<=1) {
        r = 240;
        g = 239;
        b = 181;
    } else if (srm>1 && srm<=2) {
        r = 233;
        g = 215;
        b = 108;
    } else if (srm>2) {
        // Set red decimal
        if (srm<70.6843) {        
            r = 243.8327-6.4040*srm+0.0453*srm*srm;
        } else {
            r = 17.5014;
        }
        // Set green decimal
        if (srm<35.0674) {
            g = 230.929-12.484*srm+0.178*srm*srm;
        } else {
            g = 12.0382;
        }
        // Set blue decimal
        if (srm<4) {
            b = -54*srm+216;
        } else if (srm>=4 && srm<7) {
            b = 0;
        } else if (srm>=7 && srm<9) {
            b = 13*srm-91;
        } else if (srm>=9 && srm<13) {
            b = 2*srm+8;
        } else if (srm>=13 && srm<17) {
            b = -1.5*srm+53.5;
        } else if (srm>=17 && srm<22) {
            b = 0.6*srm+17.8;
        } else if (srm>=22 && srm<27) {
            b = -2.2*srm+79.4;
        } else if (srm>=27 && srm<34) {
            b = -0.4285*srm + 31.5714;
        } else {
            b = 17;
        }
    }
    var red = doubleToHex(r);
    var green = doubleToHex(g);
    var blue = doubleToHex(b);
    return ""+red+green+blue;
}

function doubleToHex(d) {
    // Converts decimal in string to hex in string 
    var hexText = d.toString(16);
    var point = hexText.indexOf(".");
    if (point !== -1) {
        hexText = hexText.substring(0,point);
    }
    while (hexText.length < 2) {
        hexText = "0"+hexText;    
    }
    return hexText;
}