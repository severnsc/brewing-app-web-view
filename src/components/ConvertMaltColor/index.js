import React from "react"
import PropTypes from "prop-types"
import { MaltColor } from ".."
import { convertColor } from "../../utils"

const ConvertMaltColor = ({ from, to, value }) => {

	const convertedColor = convertColor(from, to, value)

	let srmValue
	if(from === "SRM"){
		srmValue = value
	}else{
		srmValue = convertColor(from, "SRM", value)
	}

	return <MaltColor srmValue={srmValue} value={convertedColor} />

}

ConvertMaltColor.propTypes = {
	from: PropTypes.oneOf(["SRM", "L", "EBC"]).isRequired,
	to: PropTypes.oneOf(["SRM", "L", "EBC"]).isRequired,
	value: PropTypes.number.isRequired
}

export default ConvertMaltColor