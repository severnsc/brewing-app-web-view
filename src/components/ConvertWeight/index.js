import React from "react"
import PropTypes from "prop-types"
import { Weight } from ".."
import { convertLbsToKg, convertKgToLbs } from "../../utils"

const ConvertWeight = ({ from, to, amount }) => {

	let convertedAmount
	if(from === "imperial" && to === "metric"){
		convertedAmount = convertLbsToKg(amount)
	}

	if(from === "metric" && to === "imperial"){
		convertedAmount = convertKgToLbs(amount)
	}

	return(
		<Weight unit={to} amount={convertedAmount} />
	)
}

ConvertWeight.propTypes = {
	from: PropTypes.oneOf(["metric", "imperial"]).isRequired,
	to: PropTypes.oneOf(["metric", "imperial"]).isRequired,
	amount: PropTypes.number.isRequired
}

export default ConvertWeight