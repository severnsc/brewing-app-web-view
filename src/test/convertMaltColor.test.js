import React from 'react';
import renderer from "react-test-renderer"
import { ConvertMaltColor } from "../components"
import { convertColor } from "../utils"
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

describe("convert malt color component", () => {

	it("renders a convert malt color component", () => {
		const convertMaltColor = renderer.create(<ConvertMaltColor from="SRM" to="L" value={1} />)
		const tree = convertMaltColor.toJSON()
		expect(tree).toMatchSnapshot()
	})

	describe("when from is SRM and to is L", () => {

		it("converts the value properly and passes to value prop", () => {
			const convertMaltColor = shallow(<ConvertMaltColor from="SRM" to="L" value={1} />)
			const convertedValue = convertColor("SRM", "L", 1)
			expect(convertMaltColor.find("MaltColor").prop("value")).toBe(convertedValue)
		})

	})

	describe("when from is L and to is SRM", () => {

		it("converts the value properly and passes to value prop", () => {
			const convertMaltColor = shallow(<ConvertMaltColor from="L" to="SRM" value={1} />)
			const convertedValue = convertColor("L", "SRM", 1)
			expect(convertMaltColor.find("MaltColor").prop("value")).toBe(convertedValue)
		})

	})

})