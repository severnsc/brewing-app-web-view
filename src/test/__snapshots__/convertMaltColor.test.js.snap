// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`convert malt color component renders a convert malt color component 1`] = `
<div
  style={
    Object {
      "background": "#e9d76c",
      "borderRadius": "20px",
      "height": "16px",
      "padding": "4px",
      "position": "relative",
    }
  }
>
  <span
    style={
      Object {
        "alignItems": "center",
        "background": "rgb(250, 250, 250)",
        "borderRadius": "50%",
        "display": "flex",
        "height": "25px",
        "justifyContent": "center",
        "left": "-5px",
        "position": "absolute",
        "top": "0px",
        "width": "25px",
      }
    }
  >
    1
  </span>
</div>
`;
